CONTRIBUTING.md

```diff
@@ -36,6 +36,38 @@ For the moment it would not seem appropriate for our plugin to throw instances c
 of the [Error class](https://api.dart.dev/stable/2.4.0/dart-core/Error-class.html), as this is perhaps more
 appropriate for programmer's failures on the Dart side of things. But time will tell.
 
+### Push Notifications 
+
+#### Debugging the app from launch
+
+- Android: Go into developer options and use the "Wait for debugger" option. Launch the app from the Flutter project (not the `android` directory) in Android Studio. Then in the Android project in Android Studio, attach the debugger.
+- iOS: In Xcode, build the Xcode app scheme with launch option: "Wait for the executable to be launched" instead of "Automatically". You need to repeat this for every launch. Then "Attach Flutter" in Android Studio or CLI if you want to, as well. 
+
+#### Push Notifications activation / deactivation
+
+The platform SDKs ([ably-android](https://github.com/ably/ably-java), [ably-cocoa](https://github.com/ably/ably-cocoa)) provide ways to check if device activation, deactivation or registration update fails. On Android, these errors are sent in Intents which you should register for at runtime. In Cocoa, your errors are provided through `ARTPushRegistererDelegate` methods. However, this error does not get returned immediately/quickly in all cases. For example, if there was no internet connection, then `Push.activate()` will not throw an error, it will just block forever, because errors are not provided by the SDKs. Once an internet connection is made, the intent will be sent / delegate methods will be called.
```

Comment:
```suggestion
The platform SDKs ([ably-android](https://github.com/ably/ably-java) and [ably-cocoa](https://github.com/ably/ably-cocoa)) enable you to check if device activation, deactivation, or registration update fails. On Android, these errors are sent in "intents" which you should register for at runtime. In Cocoa, your errors are provided through `ARTPushRegistererDelegate` methods. However, this error does not get returned immediately/quickly in all cases. For example, if there was no internet connection, then `Push.activate()` will not throw an error, it will just block forever, because errors are not provided by the SDKs. Once an internet connection is made, the intent will be sent / delegate methods will be called.
```

Reply:
Also, it's not clear (at least to me) whether the text that starts "However, this error does not get returned..." applies to both Android and Cocoa, or just the latter.

Reply:
642a0a5f5cada12e98ca0a52c809db26ad200bf5

Reply:
642a0a5f5cada12e98ca0a52c809db26ad200bf5